common/src/main/java/com/wynntils/wrappedscreens/trademarket/TradeMarketSearchResultHolder.java

```diff
@@ -0,0 +1,370 @@
+
+        // We don't want the items to be set on our custom screen
+        event.setCanceled(true);
+
+        ItemStack itemStack = event.getItemStack();
+
+        handleSetItem(slot, itemStack);
+        updateDisplayItems(wrappedScreen.getSearchQuery());
+    }
+
+    @Override
+    protected Pattern getReplacedScreenTitlePattern() {
+        return TITLE_PATTERN;
+    }
+
+    @Override
+    protected TradeMarketSearchResultScreen createWrappedScreen(WrappedScreenInfo wrappedScreenInfo) {
+        return new TradeMarketSearchResultScreen(wrappedScreenInfo, this);
+    }
+
+    @Override
+    protected void setWrappedScreen(TradeMarketSearchResultScreen wrappedScreen) {
+        this.wrappedScreen = wrappedScreen;
+
+        // Preload the first batch of pages
+        loadItemsUntilPage(PAGE_BATCH_SIZE, false);
+    }
+
+    @Override
+    protected void reset() {
+        requestedPage = -1;
+        requestedItem = null;
+        pageLoadingMode = PageLoadingMode.NONE;
+        queuedAction = null;
+        allPagesLoaded = false;
+        currentPage = 0;
+        itemMap = new HashMap<>();
+        pageItemCount = 0;
+        filteredItems = new ArrayList<>();
+
+        this.wrappedScreen = null;
+    }
+
+    public void clickOnItem(ItemStack clickedItem) {
+        // When writing this, items only update on page change,
+        // so if we don't switch pages, we can just click on the item
+        Int2ObjectMap.FastEntrySet<ItemStack> currentPageEntries =
+                itemMap.getOrDefault(currentPage, new Int2ObjectOpenHashMap<>()).int2ObjectEntrySet();
+        for (Int2ObjectMap.Entry<ItemStack> entry : currentPageEntries) {
+            if (ItemUtils.isItemEqual(entry.getValue(), clickedItem)) {
+                // Item found on the current page, click on it
+                WrappedScreenInfo wrappedScreenInfo = wrappedScreen.getWrappedScreenInfo();
+                ContainerUtils.clickOnSlot(
+                        entry.getIntKey(),
+                        wrappedScreenInfo.containerId(),
+                        GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                        wrappedScreenInfo.containerMenu().getItems());
+
+                return;
+            }
+        }
+
+        // Assume the item is before the current page
+        int pageToLoad = 0;
+
+        // Check if the item is on a later page
+        for (int i = currentPage + 1; i < itemMap.size(); i++) {
+            boolean foundItem = false;
+
+            Int2ObjectOpenHashMap<ItemStack> itemsOnPage = itemMap.get(i);
+
+            for (ItemStack itemStack : itemsOnPage.values()) {
+                if (ItemUtils.isItemEqual(itemStack, clickedItem)) {
+                    // Item found on a later page, load the pages until the last one and click on it
+                    foundItem = true;
+                    pageToLoad = itemMap.size() - 1;
+                    break;
+                }
+            }
+
+            if (foundItem) break;
+        }
+
```

Comment:
What if foundItem is false? Then you are going to call runOrQueueAction with pageToLoad == 0? Is that intended?

Reply:
Well, semi intended. We won't be able to click the item, but loading page 0 won't break the current state. I will leave this open for a bit.